src/GradientOp.jl

```diff
@@ -35,9 +35,9 @@ function GradientOp(T::Type, shape::NTuple{N,Int64}, dim::Int64) where N
   nrow = div( (shape[dim]-1)*prod(shape), shape[dim] )
   ncol = prod(shape)
   return LinearOperator{T}(nrow, ncol, false, false,
-                    x-> grad(x,shape,dim),
-                    x-> grad_t(x,shape,dim),
-                    x-> grad_t(x,shape,dim) )
+                          (res, x, α, β) -> (res .= α .* grad(x,shape,dim) ; res), 
+                          (res, x, α, β) -> (res .= α .* grad_t(x,shape,dim) .+ (β .* res); res), 
+                          (res, x, α, β) -> (res .= α .* grad_t(x,shape,dim) .+ (β .* res); res) )
```

Comment:
Not sure what the issue was, but I think you have to distinguish between the cases `β==0` and `β!=0`, because the `*` operator allocates an `res` vector that might contain `Nan`s  (cf. https://juliasmoothoptimizers.github.io/LinearOperators.jl/stable/tutorial/#Using-functions)

How about this:

```
function GradientOp(T::Type, shape::NTuple{N,Int64}, dim::Int64) where N
  nrow = div( (shape[dim]-1)*prod(shape), shape[dim] )
  ncol = prod(shape)

  function forwardGradient!(res, x, α, β)
    if β == 0
      res .= α .* grad(x,shape,dim)
    else
      res .= α .* grad(x,shape,dim) .+ β .* res
    end
    return res
  end

  function adjointGradient!(res, x, α, β)
    if β == 0
      res .= α .* grad_t(x,shape,dim)
    else
      res .= α .* grad_t(x,shape,dim) .+ β .* res
    end
    return res
  end

  return LinearOperator{T}(nrow, ncol, false, false, forwardGradient!, adjointGradient!, adjointGradient!)
end
```

How did you solve this issue for the other operators?